Automation/S3/delete_s3_buckets.py

```diff
@@ -0,0 +1,194 @@
+    '''
+    try:
+        get_bucket_tags_response = s3_client.get_bucket_tagging(
+            Bucket=bucket_name
+        )
+                    
+        bucket_owner_email_list = list(
+            filter(
+                lambda tag: tag['Key'] == "OwnerEmail",
+                get_bucket_tags_response['TagSet']
+            )
+        )
+
+        bucket_owner_email = "No Owner"
+        if len(bucket_owner_email_list) == 1:
+            bucket_owner_email = bucket_owner_email_list[0]['Value']
+
+        return bucket_owner_email
+
+    except s3_client.exceptions.from_code('NoSuchTagSetError') as e:
+        print(Fore.YELLOW + "Bucket does not have any tags!")
+
+def get_bucket_object_list(s3_client, bucket_name):
+    '''
+    Gets the names of the objects in an S3 bucket returns them as a list
+
+        Parameters:
+            s3_client (S3Client): the client necessary to execute S3 commands
+            bucket_name (str): the name of the bucket you would like the objects of
+        
+        Returns:
+            bucket_object_list (list): The list of object names
+    '''
+    try:
+        get_bucket_object_list_response = s3_client.list_objects_v2(
+            Bucket=bucket_name
+        )
+
+        bucket_object_list = []
+        if ('Contents' in get_bucket_object_list_response):
+            bucket_object_list = list(
+                map(
+                    lambda bucket_object: bucket_object['Key'],
+                    get_bucket_object_list_response['Contents']
+                )
+            )
+        else:
+            print(Fore.YELLOW + "Bucket is empty!")
+
+        return bucket_object_list
+
+    except s3_client.exceptions.from_code('NoSuchBucket') as e:
+        print(Fore.RED + "Bucket does not exist!")
+    
+
+def delete_buckets(s3_client, buckets_to_delete_list):
+    '''
+    Deletes the S3 buckets and their objects
+
+        Parameters:
+            s3_client (S3Client): the client necessary to execute S3 commands
+            buckets_to_delete_list (list): a list of bucket names and bucket objects of the form [{ 'BucketName': 'string', 'Objects': [{'Key': 'string'}] }]
+    '''
+    try:
+        for bucket_to_delete in buckets_to_delete_list:
+            print(f'Attempting to delete bucket: {bucket_to_delete["BucketName"]}...')
+
+            if len(bucket_to_delete['Objects']) > 0:
+                print("Bucket has objects, attempting to delete them...")
+                objects_deletion_response = s3_client.delete_objects(
+                    Bucket=bucket_to_delete['BucketName'],
+                    Delete={
+                        'Objects': bucket_to_delete['Objects']
+                    }
+                )
+                print(Fore.GREEN + "Objects deleted!")
+            
+            bucket_deletion_response = s3_client.delete_bucket(
+                Bucket=bucket_to_delete['BucketName']
+            )
+
+            print(Fore.GREEN + "Bucket deleted! \n")
```

Comment:
Would be nice to add information what was the name of the bucket deleted. The printed message will be much more helpful.